### PR TITLE
Use form

### PR DESCRIPTION
**Pull request by thegrinder**
First version of the hook to manage forms

### CODE REVIEW

File: src/components/hooks/useForm/useForm.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import { useState, useCallback, useMemo } from 'react';
+
+interface TDictionary<T> {
```
Comment (poteirard) [inline src/components/hooks/useForm/useForm.ts]: I think you can just use `Record` for that

File: src/components/hooks/useForm/useForm.test.ts [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+  afterEach(() => {
+    onSubmit.mockClear();
+  });
+
+  it('should return the correct object', () => {
+    const { result } = renderHook(() =>
+      useForm({
+        initialValues,
+        onSubmit,
+        validate,
```
Comment (poteirard) [inline src/components/hooks/useForm/useForm.test.ts]: there's a type error here:
```
Error:(36, 9) TS2322: Type '(values: TTestForm) => Partial<TTestForm>' is not assignable to type 'TValidate'.
  Types of parameters 'values' and 'values' are incompatible.
    Type 'TDictionary<any>' is missing the following properties from type 'TTestForm': name, lastName
```

Comment (thegrinder) [reply src/components/hooks/useForm/useForm.test.ts]: modified types and that shouldn't happen anymore. Also, I'm gonna be looking at improving them

File: src/components/hooks/useForm/useForm.test.ts [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+      }),
+    );
+    ['getFieldProps', 'invalid', 'submitting', 'handleSubmit'].forEach(property => {
```
[the remark this answers is not in the document]
Comment (thegrinder) [reply src/components/hooks/useForm/useForm.test.ts]: you're right, I'll remove the snapshot for both

File: src/components/hooks/useForm/useForm.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        onSubmit(values);
+      }
+    },
+    [invalid, onSubmit, values],
+  );
+
+  return {
+    getFieldProps,
+    invalid,
+    submitting,
```
[the remark this answers is not in the document]
Comment (thegrinder) [reply src/components/hooks/useForm/useForm.ts]: I'm probably getting ahead of myself here - thinking about manually feeding context to form provider. I'll remove that and we can think about it again once we need it

**Comment by thegrinder** [comment]
@lluia I'm gonna be building form components around this hook soon and the tests there are gonna be written in a way that you show above. I think we're aiming to have two form options: hook-only solution and set of components that use that hook internally

**Comment by lluia** [comment]
@thegrinder coolios, happy to leave tests as they're now then 💪🏻

**Pull request merged**